keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/DoubleVertexTest.java

```diff
@@ -89,4 +91,11 @@ public void canConcat() {
         DoubleVertex concatDimOne = DoubleVertex.concat(1, A, B);
         assertArrayEquals(concatDimOne.getShape(), new long[]{2, 4});
     }
+
+    @Test
+    public void canCastToIntegerVertex() {
+        DoubleVertex v1 = new ConstantDoubleVertex(3.0);
```

Comment:
It would be a more instructive test if you were to use a non-integer value (3.0). In particular,  if you chose a value between 3.5 and 4.0 then it demonstrates that it does a `floor` rather than a `round`.

Reply:
+1

Reply:
Good point, have added.